shakenfist/virt.py

```diff
@@ -471,49 +470,52 @@ def power_on(self):
                 config.parsed.get('NODE_NAME'), self.db_entry['uuid'], 'error',
                 'missing domain file in power on')
 
-        libvirt = util.get_libvirt()
-        with open(self.xml_file) as f:
-            xml = f.read()
+        with db.get_lock('virt', None, self.db_entry['uuid'], ttl=120):
```

Comment:
This should be 'instance' instead of 'virt' shouldn't it?

Reply:
The problem is that `queues.py instance_delete()` takes the `instance` lock and then eventually calls `instance.power_off()`. Cleaner also calls but without a lock. That is fine since the VM is in state `deleted` anyway.

So I chose a non-colliding name that pertained only to virt.py making power state changes.

I'll take the locks higher and use `instance` locks. `virt.py` will have to trust any callers to it that they have a lock.

Reply:
done

Reply:
So yeah, the other issue is we've talked about moving to using the object to generate the name of the lock and that wouldn't have worked here.